packages/mui-material-next/src/Button/Button.test.js

```diff
@@ -0,0 +1,383 @@
+import * as React from 'react';
+import { expect } from 'chai';
+import {
+  describeConformance,
+  act,
+  createClientRender,
+  fireEvent,
+  createServerRender,
+} from 'test/utils';
+import Button, { buttonClasses as classes } from '@mui/material/Button';
+import ButtonBase from '@mui/material/ButtonBase';
+
+describe('<Button />', () => {
+  const render = createClientRender();
+
+  describeConformance(<Button startIcon="icon">Conformance?</Button>, () => ({
+    classes,
+    inheritComponent: ButtonBase,
+    render,
+    refInstanceof: window.HTMLButtonElement,
+    muiName: 'MuiButton',
+    testDeepOverrides: { slotName: 'startIcon', slotClassName: classes.startIcon },
+    testVariantProps: { variant: 'contained', fullWidth: true },
+    testStateOverrides: { prop: 'size', value: 'small', styleKey: 'sizeSmall' },
+    skip: ['componentsProp'],
+  }));
+
+  it('should render with the root, text, and textPrimary classes but no others', () => {
+    const { getByRole } = render(<Button>Hello World</Button>);
+    const button = getByRole('button');
+
+    expect(button).to.have.class(classes.root);
+    expect(button).to.have.class(classes.text);
+    expect(button).to.have.class(classes.textPrimary);
+    expect(button).not.to.have.class(classes.textSecondary);
```

Comment:
Maybe iterate through all classes and make sure only those three are rendered?

Reply:
I just copied this file as is from @mui/material so the same tests are run for material and material-next components.